### ISSUE TITLE
Coprime missing in implementation of c_amod15 gate in Shor's Algorithm section

### ISSUE THREAD

**Issue by NikosGavalas** [qiskit-community/qiskit-textbook#1318]
**Describe the issue**

The implementation of the modulo gate used in the Shor's algorithm section is the following:

```
def c_amod15(a, power):
    """Controlled multiplication by a mod 15"""
    if a not in [2,7,8,11,13]:
        raise ValueError("'a' must be 2,7,8,11 or 13")
    U = QuantumCircuit(4)        
    for iteration in range(power):
        if a in [2,13]:
            U.swap(0,1)
            U.swap(1,2)
            U.swap(2,3)
        if a in [7,8]:
            U.swap(2,3)
            U.swap(1,2)
            U.swap(0,1)
        if a == 11:
            U.swap(1,3)
            U.swap(0,2)
        if a in [7,11,13]:
            for q in range(4):
                U.x(q)
    U = U.to_gate()
    U.name = "%i^%i mod 15" % (a, power)
    c_U = U.control()
    return c_U
```

**Describe what you expect to see**

Shouldn't it include **4** in the list of acceptable values of _a_? 4 and 15 are co-primes. Am I missing something?

It should look like this I think:

```
def c_amod15(a, power):
    """Controlled multiplication by a mod 15"""
    if a not in [2,4,7,8,11,13]:
        raise ValueError("'a' must be 2,4,7,8,11 or 13")
    U = QuantumCircuit(4)        
    for iteration in range(power):
        if a in [2,13]:
            U.swap(0,1)
            U.swap(1,2)
            U.swap(2,3)
        if a in [7,8]:
            U.swap(2,3)
            U.swap(1,2)
            U.swap(0,1)
        if a in [4, 11]:
            U.swap(1,3)
            U.swap(0,2)
        if a in [7,11,13]:
            for q in range(4):
                U.x(q)
    U = U.to_gate()
    U.name = "%i^%i mod 15" % (a, power)
    c_U = U.control()
    return c_U
```

**Comment by bopardikarsoham** [comment]
I would like to work on this issue @frankharkins

**Comment by divshacker** [comment]
```
    "        if a == 11:
    "            U.swap(1,3)
    "            U.swap(0,2)
    "        if a in [7,11,13]:
    "            for q in range(4):
    "                U.x(q)
```
I suppose it should be 

```
if a in [4, 11]:
            U.swap(1,3)
            U.swap(0,2)
        if a in [7,11,13]:
            for q in range(4):
                U.x(q)
```

**Comment by frankharkins** [comment]
Closed via #1355
